app/src/main/java/io/stormbird/wallet/service/KeystoreAccountService.java

```diff
@@ -0,0 +1,408 @@
+                    message, credentials.getEcKeyPair());
+            byte[] signed = bytesFromSignature(signatureData);
+            signed = patchSignatureVComponent(signed);
+            return signed;
+        }).subscribeOn(Schedulers.io());
+    }
+
+    //In all cases where we need to sign data the signature needs to be in Ethereum format
+    //Geth gives us the pure EC function, but for hash signing
+    @Override
+    public Single<byte[]> signTransaction(Wallet signer, byte[] message, long chainId)
+    {
+        return Single.fromCallable(() -> {
+            //byte[] messageHash = Hash.sha3(message);
+            byte[] signed = keyService.signData(signer.address, message);
+
+            signed = patchSignatureVComponent(signed);
+            return signed;
+        }).subscribeOn(Schedulers.io());
+    }
+
+    @Override
+    public boolean hasAccount(String address) {
+        address = Numeric.cleanHexPrefix(address);
+        File[] contents = keyFolder.listFiles();
+        if (contents == null) return false;
+        for (File f : contents)
+        {
+            if (f.getName().contains(address))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    @Override
+    public Single<Wallet[]> fetchAccounts() {
+        return Single.fromCallable(() -> {
+            File[] contents = keyFolder.listFiles();
+            List<Date> fileDates = new ArrayList<>();
+            Map<Date, String> walletMap = new HashMap<>();
+            List<Wallet> wallets = new ArrayList<>();
+            if (contents == null || contents.length == 0) return new Wallet[0];
+            //Wallet[] result = new Wallet[contents.length];
+            for (File f : contents)
+            {
+                String fName = f.getName();
+                int index = fName.lastIndexOf("-");
+                String address = "0x" + fName.substring(index + 1);
+                if (WalletUtils.isValidAddress(address))
+                {
+                    String d = fName.substring(5, index-1).replace("T", " ").substring(0, 23);
+                    SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH-mm-ss.SSS", Locale.ROOT);
+                    Date date = simpleDateFormat.parse(d);
+                    fileDates.add(date);
+                    walletMap.put(date, address);
+                }
+            }
+
+            Collections.sort(fileDates);
+
+            //now build a date sorted array:
+            for (Date d : fileDates)
+            {
+                String address = walletMap.get(d);
+                Wallet wallet = new Wallet(address);
+                wallet.type = WalletType.KEYSTORE;
+                wallets.add(wallet);
+            }
+
+            return wallets.toArray(new Wallet[0]);
+        })
+        .subscribeOn(Schedulers.io());
+    }
+
+    private byte[] patchSignatureVComponent(byte[] signature)
```

Comment:
Could you document this method? A little unclear what the byte manipulations below achieve.